### PR TITLE
Additional `AlternateFLH` entries beyond five

### PR DESCRIPTION
**Pull request by ststl-s**
`AlternateFLH` can have more than five.

### CODE REVIEW

File: CREDITS.md [hunk 1]
```diff
@@ -200,7 +200,9 @@ This page lists all the individual contributions to the project by their author.
 - **NetsuNegi** - Forbidding parallel AI queues by type
 - **Apollo** - Translucent SHP drawing patches
 - **FlyStar** - Campaign load screen PCX support
-- **ststl** - Use `CustomPalette` for animations with `Tiled=yes`
+- **ststl**
+   - Use `CustomPalette` for animations with `Tiled=yes`
+   - `AlternateFLH` can have more than five
```
Comment (Starkku) [inline CREDITS.md:205]: Needs spelling tweaks.

Suggestion: Additional `AlternateFLH` entries beyond five

File: docs/Fixed-or-Improved-Logics.md [hunk 2]
```diff
@@ -89,6 +89,7 @@ This page describes all ingame logics that are fixed or improved in Phobos witho
 - IvanBomb images now display and the bombs detonate at center of buildings instead of in top-leftmost cell of the building foundation.
 - Fixed BibShape drawing for a couple of frames during buildup for buildings with long buildup animations.
 - Animation with `Tiled=yes` now supports `CustomPalette`.
+- `AlternateFLH` can have more than five.
```
Comment (Starkku) [inline docs/Fixed-or-Improved-Logics.md:92]: Needs spelling tweaks.

Suggestion: More than five entries are now read for `AlternateFLH`.

File: docs/Whats-New.md [hunk 3]
```diff
@@ -254,6 +254,7 @@ New:
 - Forcing specific weapon against cloaked or disguised targets (by Starkku)
 - Customizable ROF random delay (by Starkku)
 - Animation with `Tiled=yes` now supports `CustomPalette` (by ststl)
+- `AlternateFLH` can have more than five (by ststl)
```
Comment (Starkku) [inline docs/Whats-New.md:257]: Needs spelling tweaks.

Suggestion: More than five entries are now read for `AlternateFLH` (by ststl)

File: src/Ext/Techno/Hooks.Firing.cpp [hunk 4]
```diff
@@ -409,3 +409,23 @@ DEFINE_HOOK(0x6FF4CC, TechnoClass_FireAt_ToggleLaserWeaponIndex, 0x6)
...
+	const CoordStruct& flh =
+		weaponIdx < static_cast<int>(pTypeExt->AlternateFLHs.size())
+		? pTypeExt->AlternateFLHs[weaponIdx]
+		: CoordStruct::Empty;
+
+	R->ECX(flh.X);
+	R->EBP(flh.Y);
+	R->EAX(flh.Z);
+
+	return 0x6F3B37;
```
Comment (Starkku) [inline src/Ext/Techno/Hooks.Firing.cpp:430]: Use anonymous enum for non-zero return addresses with appropriate name.

E.g `enum { SkipGameCode = 0x6F3B37 };`, which is a commonly used label in existing code mostly by yours truly for cases where a hook simply replaces a portion of game code.

**Review by Starkku — CHANGES_REQUESTED** [review]
Some minor nitpicks but otherwise looks alright.

File: src/Ext/Techno/Hooks.Firing.cpp (hunk 4 above)
Comment (Metadorius) [reply src/Ext/Techno/Hooks.Firing.cpp:430]: Well in such cases I think this is not strictly necessary, as one-addr-return-only hooks are pretty self-explanatory IMO, but I don't mind personally.

Comment (Starkku) [reply src/Ext/Techno/Hooks.Firing.cpp:430]: Fair enough, there's been some inconsistency on if it is needed or not in the past which explains my stance.

File: src/Ext/TechnoType/Body.cpp [hunk 5]
```diff
@@ -254,6 +254,21 @@ void TechnoTypeExt::ExtData::LoadFromINIFile(CCINIClass* const pINI)
...
 	TechnoTypeExt::GetBurstFLHs(pThis, exArtINI, pArtSection, CrouchedWeaponBurstFLHs, EliteCrouchedWeaponBurstFLHs, "Prone");
 
+	for (size_t i = 0;; i++)
+	{
+		char key[0x39];
+		Nullable<CoordStruct> alternateFLH;
+		sprintf_s(key, "AlternateFLH%u", i);
+		alternateFLH.Read(exArtINI, pArtSection, key);
+
+		if (i >= 5U && !alternateFLH.isset())
```
Comment (Metadorius) [inline src/Ext/TechnoType/Body.cpp:264]: Maybe needs a comment explaining why we always check the first five elements.

**Pull request merged**